core/ledger/kvledger/txmgmt/privacyenabledstate/db.go

```diff
@@ -124,12 +130,13 @@ func (p *DBProvider) Drop(ledgerid string) error {
 type DB struct {
 	statedb.VersionedDB
 	metadataHint *metadataHint
+	localMspId   string
 }
 
 // NewDB wraps a VersionedDB instance. The public data is managed directly by the wrapped versionedDB.
 // For managing the hashed data and private data, this implementation creates separate namespaces in the wrapped db
 func NewDB(vdb statedb.VersionedDB, ledgerid string, metadataHint *metadataHint) (*DB, error) {
-	return &DB{vdb, metadataHint}, nil
+	return &DB{vdb, metadataHint, viper.GetString("peer.localMspId")}, nil
```

Comment:
Getting the config from within the implementation is considered a bad practice. We use the dependency injection pattern to pass down dependencies including peer configuration.
In this case, in kv_ledger_provider.go when calling privacyenabledstate.NewDBProvider() you can pass the existing 
 `p.initializer.MembershipInfoProvider.MyImplicitCollectionName()`.